source/beamline/SGM/SGMBeamline.h

```diff
@@ -23,490 +23,137 @@ along with Acquaman.  If not, see <http://www.gnu.org/licenses/>.
 #define AM_SGMBEAMLINE_H
 
 #include "beamline/CLS/CLSBeamline.h"
-#include "dataman/SGM/SGMBeamlineInfo.h"
 
-#include "beamline/AMDetectorGroup.h"
-#include "beamline/AMDetectorSet.h"
-#include "beamline/AMControlSet.h"
-#include "util/AMBiHash.h"
-#include "beamline/CLS/CLSSIS3820Scaler.h"
-#include "beamline/AMControlSetSampleManipulator.h"
-#include "beamline/CLS/CLSSynchronizedDwellTime.h"
-
-#define SGMBEAMLINE_PV_NAME_LOOKUPS_FAILED 312001
-
-class AMSamplePlatePre2013;
 class SGMMAXvMotor;
 class AMMotorGroup;
-
-class CLSAmptekSDD123DetectorNew;
-class CLSPGTDetectorV2;
-class CLSQE65000Detector;
 class CLSAdvancedScalerChannelDetector;
-class AMBasicControlDetectorEmulator;
-
+/*!
+  * A singleton class which represents the SGM beamline. The beamline class can
+  * be accessed through the SGMBeamline::sgm() function.
+  */
 class SGMBeamline : public CLSBeamline
 {
 Q_OBJECT
 public:
 
-	static SGMBeamline* sgm();		// singleton-class accessor
+	/*!
+	  * Singletone class accessor. Provides access to the SGMBeamline class
+	  * instance.
+	  */
+	static SGMBeamline* sgm();
+
+	/*!
+	  * Virtual destructor for the SGMBeamline class.
+	  */
 	virtual ~SGMBeamline();
 
-	bool isConnected() const;
```

Comment:
It would probably be worth it to implement the `isConnected()` method and provide a signal for it.  We can worry about other forms of "Ready" as we need them.